### ISSUE TITLE
Permission denied - Unable to start Kestrel

### ISSUE THREAD

**Issue by dakipro** [yllibed/Zigbee2MqttAssistant#91]
Hi,
Thanks for the project. I am trying to run it as a docker container on ubuntu 18 server, using ansible to deploy it. 
My parameters for container are 
```

- name: ensure zigbee2mqttassistant is started
  docker_container:
    name: zigbee2mqttassistant
    image: "carldebilly/zigbee2mqttassistant"
    state: started
    restart_policy: always
    detach: true
    user: "mainuser"
    recreate: yes
    published_ports:
      - "8881:80"
    env:
        Z2MA_SETTINGS__MQTTSERVER: "192.168.1.1"
        Z2MA_SETTINGS__MQTTUSERNAME: "abc"
        Z2MA_SETTINGS__MQTTPASSWORD: "def"
```

However it does not start properly, it complains that 

`
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)

   at Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions.BindAsync(AddressBindContext context)

   at Microsoft.AspNetCore.Server.Kestrel.Core.AnyIPListenOptions.BindAsync(AddressBindContext context)

   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.AddressesStrategy.BindAsync(AddressBindContext context)

   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(IServerAddressesFeature addresses, KestrelServerOptions serverOptions, ILogger logger, Func`2 createBinding)

   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)

   at Microsoft.AspNetCore.Hosting.Internal.WebHost.StartAsync(CancellationToken cancellationToken)

   at Microsoft.AspNetCore.Hosting.WebHostExtensions.RunAsync(IWebHost host, CancellationToken token, String shutdownMessage)

   at Microsoft.AspNetCore.Hosting.WebHostExtensions.RunAsync(IWebHost host, CancellationToken token)

   at Microsoft.AspNetCore.Hosting.WebHostExtensions.Run(IWebHost host)

   at Zigbee2MqttAssistant.Program.Main(String[] args) in d:\a\1\s\Zigbee2MqttAssistant\Program.cs:line 18

crit: Microsoft.AspNetCore.Server.Kestrel[0]

      Unable to start Kestrel.

System.Net.Sockets.SocketException (13): Permission denied

   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)

   at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)

   at System.Net.Sockets.Socket.Bind(EndPoint localEP)

   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransport.BindAsync()

   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.<>c__DisplayClass21_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()

--- End of stack trace from previous location where exception was thrown ---
`

I tried several combination of paramters, restarting the host, restartint zigbee2mqtt assistant (which has homeassistant to true) etc, but no luck so far. 

As far as I understood the container does not need mapped volume to the host?
Any suggestions where I can start tweaking things around to get it to run?

**Comment by dakipro** [comment]
Update, i tried to use "carldebilly/zigbee2mqttassistant:linux-x64" and now I get this error

Did you mean to run dotnet SDK commands? Please install dotnet SDK from:
  https://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409

However, I also removed linux-x64 part now and the error is still here. Is this perhaps related to the last push?
How to get the last known stable version of docker?

Thanks

**Comment by carldebilly** [comment]
I just tried the latest version from docker hub and it works.

```
C:\src\Zigbee2MqttAssistant>docker run carldebilly/zigbee2mqttassistant:linux-x64
Unable to find image 'carldebilly/zigbee2mqttassistant:linux-x64' locally
linux-x64: Pulling from carldebilly/zigbee2mqttassistant
80369df48736: Pull complete                                                                                             5daec2aaeb45: Pull complete                                                                                             a386e5a70381: Pull complete                                                                                             0b833be3ecdc: Pull complete                                                                                             3c28df2739c6: Pull complete                                                                                             a6d7d0f959bb: Pull complete                                                                                             Digest: sha256:74d3eeb591a974120bbabec4516f95ca4cc0a20c33a1d380e4082c7e275936f6
Status: Downloaded newer image for carldebilly/zigbee2mqttassistant:linux-x64
warn: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[35]
      No XML encryptor configured. Key {ad8501a8-87f1-45bf-b9eb-b84638688a99} may be persisted to storage in unencrypted form.
warn: Zigbee2MqttAssistant.Services.SettingsService[0]
      Section 'settings' in configuration does not exists. Will use default settings instead.
Hosting environment: Production
Content root path: /app
Now listening on: http://[::]:80
Application started. Press Ctrl+C to shut down.
```

I'm on Windows, but it's using a linux VM to run it.  I also tried on a "pure" linux machine and it works too.  That's very strange.

The closest information I can see about you problem is this issue: https://github.com/dotnet/dotnet-docker/issues/765

Carl.

**Comment by dakipro** [comment]
Thank you for testing. I am happy to say that somehow I managed to make it run. Not quite sure what I did, but it was a combination of appending :linux-x64 on the name, removing container few times and probably removing "    user: "mainuser" " parameter from ansible task. 

Thanks for checking and thank you for creating this tool, it simplifies managing mqtt devices a lot. 

Without it I would have probably just purchased conbee dongle since it comes with some sort of admin (if I understood it correctly).
